src/services/outliningElementsCollector.ts

```diff
@@ -1,227 +1,167 @@
 /* @internal */
 namespace ts.OutliningElementsCollector {
-    const collapseText = "...";
-    const maxDepth = 40;
-    const defaultLabel = "#region";
-    const regionMatch = new RegExp("^\\s*//\\s*#(end)?region(?:\\s+(.*))?$");
-
     export function collectElements(sourceFile: SourceFile, cancellationToken: CancellationToken): OutliningSpan[] {
-        const elements: OutliningSpan[] = [];
-        let depth = 0;
-        const regions: OutliningSpan[] = [];
+        const res: OutliningSpan[] = [];
+        addNodeOutliningSpans(sourceFile, cancellationToken, res);
+        addRegionOutliningSpans(sourceFile, res);
+        return res.sort((span1, span2) => span1.textSpan.start - span2.textSpan.start);
+    }
 
-        walk(sourceFile);
-        gatherRegions();
-        return elements.sort((span1, span2) => span1.textSpan.start - span2.textSpan.start);
+    function addNodeOutliningSpans(sourceFile: SourceFile, cancellationToken: CancellationToken, out: Push<OutliningSpan>): void {
+        let depth = 0;
+        const maxDepth = 39;

```

Comment:
Any particular reason we changed from 40 to 39?

Reply:
Before we counted the SourceFile as depth 1 while now it's depth 0.
I've bumped this back up to 40 so it's a nice round number and changed tests to match.